tests/methods/gasscf-dsrg-mrpt2-1/input.dat

```diff
@@ -35,6 +35,7 @@ set forte {
   root_sym 0
   charge 0
   sci_enforce_spin_complete true
+  spin_project_full true
```

Comment:
In the future this option needs to be gone. If one does GASCI or some other form of truncated CI the code should always guarantee the correct spin state. It should not be dependent on some option that a new user is certainly not going to know to activate. This can be done in a future PR.

Reply:
I can enforce it to be true if doing GAS. The only question is that do we want this for selective GAS in the future? I can tighten the test to do GAS and sigma = 0.0. Let me know your thoughts. @fevangelista @huangm29

Reply:
In the C++ side, I enforce `SPIN_PROJECT_FULL` being `true` for a non-selective GASCI computation. See 30da37d.

Reply:
OK, I misread your comment. Yes, I agree that selected CI should always guarantee the correct spin state. This is an easy fix on the coding side (set the default option value to `True`). However, there will be a lot of reference values changes to ACI tests.

Reply:
@lcyyork : you are right. The distinction between selected vs. complete GAS spaces is important. But for a complete GAS this should always be set to true. I think the solution will be to have two implementations of GAS: selected and complete.